src/datasets/tsn_dataset.py

```diff
@@ -123,12 +126,26 @@ def _get_frame_idxs(
                     p += 1
         return seg_idxs
 
-    def _remove_non_scalar_values(self, metadata: Dict[str, Any]) -> Dict[str, Any]:
-        def is_scalar(v: Any):
+    def _drop_non_scalar_metadata(self, dataset: VideoDataset):
+        def not_scalar(v: Any):
             if isinstance(v, (tuple, list)):
-                return False
+                return True
             if v is None:
-                return False 
-            return True
-
-        return {k: v for k, v in metadata.items() if is_scalar(v)}
+                return True
+            return False
```

Comment:
These 5 lines can be collapsed to  `return isinstance(v, (tuple, list)) or v is None`